File: frontend/src/app/employees/services/employee.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';
import { Employee, CreateEmployeeRequest } from '../models/employee.models';

@Injectable({ providedIn: 'root' })
export class EmployeeService {

  private readonly apiUrl = `${environment.apiUrl}/employees`;

  constructor(private http: HttpClient) {}

  findAll(): Observable<Employee[]> {
    return this.http.get<Employee[]>(this.apiUrl);
  }

  findById(id: string): Observable<Employee> {
    return this.http.get<Employee>(`${this.apiUrl}/${id}`);
  }

  create(request: CreateEmployeeRequest): Observable<Employee> {
    return this.http.post<Employee>(this.apiUrl, request);
  }

  delete(id: string): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }
}

File: backend/src/main/java/com/staffengagement/employee/service/EmployeeService.java
package com.staffengagement.employee.service;

import com.staffengagement.employee.dto.CreateEmployeeRequest;
import com.staffengagement.employee.dto.EmployeeResponse;
import com.staffengagement.employee.dto.UpdateEmployeeRequest;

import java.util.List;
import java.util.UUID;

public interface EmployeeService {
    List<EmployeeResponse> findAll();
    EmployeeResponse findById(UUID id);
    boolean existsById(UUID id);
    EmployeeResponse create(CreateEmployeeRequest request);
    EmployeeResponse update(UUID id, UpdateEmployeeRequest request);
    void delete(UUID id);
}
